lib/bazel/options.go

```diff
@@ -73,17 +85,27 @@ func (o *queryOptions) filterError(err error) error {
 		return nil
 	}
 
-	if err, ok := err.(*exec.ExitError); ok {
+	var execErr *exec.ExitError
+	if errors.As(err, &execErr) {
 		// PARTIAL_ANALYSIS_FAILURE is expected when --keep_going is passed
 		// https://github.com/bazelbuild/bazel/blob/86409b7a248d1cb966268451f9aa4db0763c3eb2/src/main/java/com/google/devtools/build/lib/util/ExitCode.java#L38
-		if err.ExitCode() == 3 {
+		if execErr.ExitCode() == 3 {
```

Comment:
Why is only error code 3 expected here? Does bazel return other error codes while querying that mean other things?

Reply:
Yes; see the URL in the comment for all of bazel's return codes and their various meanings.